File: Examples/CSharp/AsposePDF/Images/PageToTIFF.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Devices;

namespace CSharp.AsposePdf.Images
{
    public class PageToTIFF
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdf_Images();

            //Open document
            Document pdfDocument = new Document(dataDir+ "PageToTIFF.pdf");

            //Create Resolution object
            Resolution resolution = new Resolution(300);

            //Create TiffSettings object
            TiffSettings tiffSettings = new TiffSettings();
            tiffSettings.Compression = CompressionType.None;
            tiffSettings.Depth = ColorDepth.Default;
            tiffSettings.Shape = ShapeType.Landscape;
            tiffSettings.SkipBlankPages = false;

            //Create TIFF device
            TiffDevice tiffDevice = new TiffDevice(resolution, tiffSettings);


            //Convert a particular page and save the image to stream
            tiffDevice.Process(pdfDocument, 1, 1, dataDir + "output.tif");
            
        }
    }
}

File: Examples/VisualBasic/AsposePDF/Images/PageToTIFF.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Devices

Namespace AsposePDF.Images
    Public Class PageToTIFF
        Public Shared Sub Run()
            ' ExStart:PageToTIFF
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdf_Images()

            ' Open document
            Dim pdfDocument As New Document(dataDir & Convert.ToString("PageToTIFF.pdf"))

            ' Create Resolution object
            Dim resolution As New Resolution(300)

            ' Create TiffSettings object
            Dim tiffSettings As New TiffSettings()
            tiffSettings.Compression = CompressionType.None
            tiffSettings.Depth = ColorDepth.[Default]
            tiffSettings.Shape = ShapeType.Landscape
            tiffSettings.SkipBlankPages = False

            ' Create TIFF device
            Dim tiffDevice As New TiffDevice(resolution, tiffSettings)

            ' Convert a particular page and save the image to stream
            tiffDevice.Process(pdfDocument, 1, 1, dataDir & Convert.ToString("PageToTIFF_out_.tif"))
            ' ExEnd:PageToTIFF
            System.Console.WriteLine("PDF one page converted to tiff successfully!")
        End Sub
    End Class
End Namespace

